kale/loaddata/video_dataset.py

```diff
@@ -0,0 +1,244 @@
+
+        Args:
+            record: VideoRecord denoting a video sample.
+        Returns:
+             List of indices of segment center frames.
+        """
+        if record.num_frames > self.num_segments + self.frames_per_segment - 1:
+            offsets = self._get_test_indices(record)
+
+        # edge case for when a video does not have enough frames
+        else:
+            offsets = np.zeros((self.num_segments,)) + 1
+        return offsets
+
+    def _get_test_indices(self, record):
+        """
+        For each segment, finds the center frame index.
+
+        Args:
+            record: VideoRecord denoting a video sample
+        Returns:
+            List of indices of segment center frames.
+        """
+
+        tick = (record.num_frames - self.frames_per_segment + 1) / float(self.num_segments)
+
+        offsets = np.array([int(tick / 2.0 + tick * x) for x in range(self.num_segments)])
+
+        return offsets + 1
+
+    def __getitem__(self, index):
+        """
+        For video with id index, loads self.NUM_SEGMENTS * self.FRAMES_PER_SEGMENT
+        frames from evenly chosen locations.
+
+        Args:
+            index: Video sample index.
+        Returns:
+            a list of PIL images or the result
+            of applying self.transform on this list if
+            self.transform is not None.
+        """
+        record = self.video_list[index]
+
+        if not self.test_mode:
+            segment_indices = self._sample_indices(record) if self.random_shift else self._get_val_indices(record)
+        else:
+            segment_indices = self._get_test_indices(record)
+
+        return self._get(record, segment_indices)
+
+    def _get(self, record, indices):
+        """
+        Loads the frames of a video at the corresponding
+        indices.
+
+        Args:
+            record: VideoRecord denoting a video sample.
+            indices: Indices at which to load video frames from.
+        Returns:
+            1) A list of PIL images or the result
+            of applying self.transform on this list if
+            self.transform is not None.
+            2) An integer denoting the video label.
+        """
+
+        images = list()
+        for seg_ind in indices:
+            frame_index = int(seg_ind)
+            for i in range(self.frames_per_segment):
+                seg_img = self._load_image(record.path, frame_index)
+                images.extend(seg_img)
+                if frame_index < record.num_frames:
+                    frame_index += 1
+
+        if self.transform is not None:
+            images = self.transform(images)
+
+        return images, record.label
+
+    def __len__(self):
+        return len(self.video_list)
+
+def imglist_totensor(img_list):
+    """
+    Converts each PIL image in a list to
+    a torch Tensor and stacks them into
+    a single tensor. Can be used as first transform
+    for ``kale.loaddata.video_dataset.VideoFrameDataset``.
```

Comment:
I initially felt like it would be better to have this specific transform in here but have changed my mind and agree that putting it into prepdata is better. Will make the change

Reply:
Yeah, as long as it is **general** and highly **reusable**, we should put it in core.